documentation/getstarted/advanced/migrationtesting.md

```diff
@@ -0,0 +1,241 @@
+
+    <pre class="console"><span>&gt;</span> export PGPASSWORD=&lt;Password&gt;</pre>
+    <pre class="console"><span>&gt;</span> pg_dump -h &lt;Host&gt; -p &lt;Port&gt; -U &lt;Username&gt; --file backups/pagila.sql</pre>
+
+1. Using Spawn, we can turn this into a [data image](https://www.spawn.cc/docs/concepts-data-image) from our backup file, following the instructions [here](https://www.spawn.cc/docs/source-configuration-backup-postgres). First, we will create a source file `pagila-backup.yaml` **in the directory which contains the `backups` folder** with the following content:
+
+    ```
+    name: Pagila
+    engine: postgresql
+    version: 12.0
+    sourceType: backup
+    backup:
+      folder: ./backups/
+      file: pagila.sql
+      format: plain
+    tags:
+      - prod
+      - latest
+    ```
+
+1. The source file describes where the data and schema comes from. We can now create a data image from this:
+
+    <pre class="console"><span>&gt;</span> spawnctl create data-image --file ./pagila-backup.yaml</pre>
+
+This set up will allow us to create databases from the backup file - we'll come back to this later on in the tutorial.
+
+## Write a GitHub Actions workflow
+
+Lets create and set up our GitHub Actions workflow to test database migrations.
+
+### Set up access token for Spawn
+
+1. To access `spawnctl` in the pipeline, we will first need to create an [access token](https://spawn.cc/docs/spawnctl-accesstoken-create#tutorial) so that we won't be prompted to authenticate with Spawn in our CI environment:
+
+    <pre class="console"><span>&gt;</span> spawnctl create access-token --purpose "CI system Spawn access token"
+    Access token generated: &lt;access-token-string&gt;</pre>
+
+1. Copy the `<access-token-string>` into your clipboard; you won't be able to retrieve it again. Now lets add it as a secret variable called `SPAWNCTL_ACCESS_TOKEN` in GitHub Actions by following this [tutorial](https://docs.github.com/en/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-a-repository).
+
+### Create GitHub workflow
+
+1. In your repository, add a file underneath the folder `.github/workflows` called `migration-test.yml`
+
+1. Copy and paste the yaml below, then commit and push the new file:
+
+    ```
+    name: Database migration test
+
+    on: workflow_dispatch
+
+    jobs:
+      run_migration_test:
+        name: Run Flyway migration tests
+        runs-on: ubuntu-latest
+        steps:
+          - name: Checkout
+            uses: actions/checkout@v2
+          - name: Run database migrations
+            run: migration-test-prod.sh
+            env:
+              SPAWNCTL_ACCESS_TOKEN: ${{ secrets.SPAWNCTL_ACCESS_TOKEN }} 
+    ```
+
+1. This workflow is referencing a bash script `migration-test-prod.sh` which can be called from any CI pipeline as it is, and just work. Lets go ahead and create that script by copying the code below and saving the file at the root level of our directory:
+
+    ```
```

Comment:
Putting `bash` after the three backticks should give bash highlighting here

Reply:
db7c314